torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1615,7 +1612,10 @@ std::vector<Val*> Index::getGlobalProducerStridedIndices(
         loops,
         root_ind);
 
-    root_ind = getProducerIndexWithHalo(producer_tv, i, root_ind, consumer_tv);
+    if (root_dom[i] != selected_id) {
```

Comment:
There is an assert inside `getProducerHaloOffset`:
```C++
  auto it = p2c.find(producer_id);
  // p2c should always have a mapping for producer_id. The only case
  // where no mapping exists for a producer axis is when it is a
  // reduction axis. Since this function is only used for indexing
  // producer tensors, where reduction axes are skipped, producer_id
  // should never be a reduction axis.
  TORCH_INTERNAL_ASSERT(it != p2c.end());
```
which fails for the select id. Do we want to remove this assert and return `0` if not mapped?

Reply:
Yeah, I think we should remove the assert, but I'd feel more assured with some validation that the producer ID is a selected ID.

Reply:
I moved the check inside `getProducerIndexWithHalo`, something like
```C++
if (override) {
  offset = 0;
} else {
  assert 
}
```